core/src/main/java/com/amplifyframework/core/model/AuthRule.java

```diff
@@ -139,29 +185,40 @@ public boolean equals(Object object) {
 
         AuthRule authRule = (AuthRule) object;
 
-        return authStrategy == authRule.authStrategy &&
+        return ObjectsCompat.equals(authStrategy, authRule.authStrategy) &&
```

Comment:
This really doesn't matter too much since `==` and `equals` do behave the same way when comparing enum values - however, `==` has a slight benefit because it does type checking at _compile time_, whereas `equals` doesn't do it until _runtime_.